src/Store/QuickmenuStoreConnector.re

```diff
@@ -230,53 +232,31 @@ let start = (themeInfo: Model.ThemeInfo.t) => {
 
     | ListFocusUp => (
         Option.map(
-          (state: Quickmenu.t) => {
-            let count = Array.length(state.items);
-
+          (state: Quickmenu.t) =>
             {
               ...state,
               focused:
-                Option.map(
-                  focused =>
-                    if (count == 0) {
-                      0;
-                    } else if (focused <= 0) {
-                      count - 1; // "roll over" to end of list
-                    } else {
-                      focused - 1;
-                    },
+                IndexEx.prevRollOverOpt(
                   state.focused,
-                )
-                |> Option.value(~default=max(count - 1, 0))  // default to end of list
-                |> Option.some,
-            };
-          },
+                  ~last=Array.length(state.items) - 1,
+                ),
+            },
           state,
         ),
         Isolinear.Effect.none,
       )
 
     | ListFocusDown => (
         Option.map(
-          (state: Quickmenu.t) => {
-            let count = Array.length(state.items);
-
+          (state: Quickmenu.t) =>
             {
               ...state,
               focused:
-                Option.map(
-                  focused =>
-                    if (count == 0) {
-                      0;
-                    } else {
-                      (focused + 1) mod count;
-                    },
+                IndexEx.nextRollOverOpt(
```

Comment:
Good idea to abstract this logic into the `IndexEx` utilities

Reply:
Not entirely sure where to put them though. They operate on index-like values, but aren't actually `Index.t`s, sp `IndexEx` is a bit misleading. And the name is a bit kludgy in general. It's Gooed Enough For Now :tm: though.